core/controllers/access_validators.py

```
@@ -441,17 +441,26 @@ def _validate_node_id(
+            arc_id: str. The 1-based arc position or full arc ID to validate.
 
         Raises:
             NotFoundException. The arc ID was not found.
         """
         all_arcs = self._get_all_arcs_for_topic(topic)
+        if arc_id.isdigit():
+            arc_index = int(arc_id)
+            if 1 <= arc_index <= len(all_arcs):
+                return
+            raise self.NotFoundException(
+                'Arc with id %s is not part of this topic.' % arc_id
```

Ruff (0.16.1)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)